internal/provider/data_source.go

```diff
@@ -5,77 +5,94 @@ import (
-				Type:        schema.TypeMap,
+				Type:        types.MapType{ElemType: types.StringType},
 				Optional:    true,
 			},
 			"outputs": {
 				Description: "After the data source is \"read\", a copy of the `inputs` map.",
-				Type:        schema.TypeMap,
+				Type:        types.MapType{ElemType: types.StringType},
 				Computed:    true,
 			},
 			"random": {
 				Description: "A random value. This is primarily for testing and has little practical use; prefer the [hashicorp/random provider](https://registry.terraform.io/providers/hashicorp/random) for more practical random number use-cases.",
-				Type:        schema.TypeString,
+				Type:        types.StringType,
 				Computed:    true,
 			},
 			"has_computed_default": {
 				Description: "If set, its literal value will be stored and returned. If not, its value defaults to `\"default\"`. This argument exists primarily for testing and has little practical use.",
-				Type:        schema.TypeString,
+				Type:        types.StringType,
 				Optional:    true,
 				Computed:    true,
 			},
 
 			"id": {
-				Description: "This attribute is only present for some legacy compatibility issues and should not be used. It will be removed in a future version.",
-				Computed:    true,
-				Deprecated:  "This attribute is only present for some legacy compatibility issues and should not be used. It will be removed in a future version.",
-				Type:        schema.TypeString,
+				Description:        "This attribute is only present for some legacy compatibility issues and should not be used. It will be removed in a future version.",
+				Computed:           true,
+				DeprecationMessage: "This attribute is only present for some legacy compatibility issues and should not be used. It will be removed in a future version.",
+				Type:               types.StringType,
 			},
 		},
-	}
+	}, nil
 }
 
-func dataSourceRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func (n nullDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var config nullDataSourceModelV0
 
-	inputs := d.Get("inputs")
-	err := d.Set("outputs", inputs)
-	if err != nil {
-		return diag.Errorf("error setting outputs: %s", err)
+	diags := req.Config.Get(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
 	}
 
-	err = d.Set("random", fmt.Sprintf("%d", rand.Int()))
-	if err != nil {
-		return diag.Errorf("error setting random int: %s", err)
-	}
-	if d.Get("has_computed_default") == "" {
-		err := d.Set("has_computed_default", "default")
-		if err != nil {
-			return diag.Errorf("error setting has_computed_default: %s", err)
-		}
-	}
+	config.Outputs = config.Inputs
+	config.Random = types.String{Value: fmt.Sprintf("%d", rand.Int())}
 
-	d.SetId("static")
+	if config.HasComputedDefault.IsNull() {
+		config.HasComputedDefault.Null = false
+		config.HasComputedDefault = types.String{Value: "default"}
```

Comment:
Nit: Since the second line overwrites the change of the first line, the first line can be removed. 👍 

```suggestion
		config.HasComputedDefault = types.String{Value: "default"}
```